include/random_quda.h

```diff
@@ -19,37 +21,47 @@ typedef struct curandStateMRG32k3a cuRNGState;
    @brief Class declaration to initialize and hold CURAND RNG states
 */
 class RNG {
-public:
-    RNG(int rng_sizes, int seedin, const int XX[4]);
-    /*! free array */
-    void Release();
-    /*! initialize curand rng states with seed */
-    void Init();
-    /*! @brief return curand rng array size */
-    int Size(){ return rng_size;};
-    int Node_Offset(){ return node_offset;};
-    int Seed(){ return seed;};
-    __host__ __device__ __inline__ cuRNGState* State(){ return state;};
-    /*! @brief Restore CURAND array states initialization */
-    void restore();
-    /*! @brief Backup CURAND array states initialization */
-    void backup();
-private:
-    /*! array with current curand rng state */
-    cuRNGState *state;
-    /*! array for backup of current curand rng state */
-    cuRNGState *backup_state;
-    /*! initial rng seed */
-    int seed;
-    /*! @brief number of curand states */
-    int rng_size;
-    /*! @brief offset in the index, in case of multigpus */
-    int node_offset;
-    int X[4];
-    /*! @brief allocate curand rng states array in device memory */
-    void AllocateRNG();
-    /*! @brief CURAND array states initialization */
-    void INITRNG(int rng_sizes, int seedin, int offsetin);
+
+  private:
+  cuRNGState *state;        /*! array with current curand rng state */
+  cuRNGState *backup_state; /*! array for backup of current curand rng state */
+  int seed;                 /*! initial rng seed */
```

Comment:
above the seed is long. Just checking whether this should be int here.

Reply:
Good question.  This is code that @nmrcardoso wrote a long time ago.  Let me see if I can work this out.....

Reply:
Ok, the interface to `curand_init` is 
```
__device__ void curand_init (unsigned long long seed, unsigned long long sequence,
                             unsigned long long offset, curandState_t *state);
```
So I think all the seeds that we carry around should in fact be `unsigned long long`, and neither `int` or `long`.  I'll fix this, including at the interface level.